_pytest/main.py

```diff
@@ -476,9 +488,39 @@ def _parsearg(self, arg):
                     " (missing __init__.py?)")
             else:
                 raise UsageError("file not found: " + arg)
+        if self.line:
+            self.nearest_func_name = self._get_nearest_func_name(self.line, path)
         parts[0] = path
         return parts
 
+    def _get_nearest_func_name(self, line, path):
+        path = str(path)
+        start_line = line = int(line) - 1
+        if not os.path.isfile(path):
+            raise UsageError("{} is not a file".format(path))
+        with open(path) as f:
```

Comment:
Hmm tricky, this might blowup depending on the encoding of the file and of the file system... we probably need to use something else to properly handle the special encoding comment that might appear at the top of the file. I wonder if we can use something in `_pytest.code` for that.

Reply:
Fixed